frontend/ui/widget/imageviewer.lua

```diff
@@ -713,42 +735,87 @@ function ImageViewer:onSpread(_, ges)
     if self._image_wg then
         self._center_x_ratio, self._center_y_ratio = self._image_wg:getPanByCenterRatio(ges.pos.x - Screen:getWidth()/2, ges.pos.y - Screen:getHeight()/2)
     end
-    -- Set some zoom increase value from pinch distance.
-    -- Making it relative to the smallest dimension between the currently scaled image or the Screen makes it less annoying
-    -- when approaching both very small scale factors (where the image dimensions are many times smaller than the screen),
+    -- We'll zoom slightly differently depending on what the gesture looks like, relative to the current size of the image:
+    -- If, at the end of the gesture, the distance between the two fingers is larger than the image's dimension,
+    -- (e.g., we've spread outward, and fingers are now outside of the image's bounds),
+    -- we just jump directly to making said dimension match said distance,
+    -- with the intent of making the image dimensions "snap" to the fingers.
+    -- We shave 10% off the gesture span, hoping that it'll make for a less frustrating experience when close to the image dims,
+    -- and to avoid potentially triggering a small zoom-in with a spread, or a small zoom-out with a pinch ;).
+    -- Otherwise, we compute a scaling percentage (which will *modify* the current scaling factor),
+    -- based on the gesture distance (it's the sum of the travel of both fingers).
+    -- In this last case, making this distance relative to the smallest dimension between
+    -- the currently scaled image or the Screen makes it less annoying when approaching both very small scale factors
+    -- (where the image dimensions are many times smaller than the screen),
     -- meaning using the image dimensions here takes less zoom steps to get it back to a sensible size;
     -- *and* large scale factors (where the image dimensions are larger than the screen),
     -- meaning using the screen dimensions here makes zoom steps, again, slightly more potent.
-    local inc
+    -- Note that, in the first case, you're very likely to end up in the first branch instead (i.e., the "snap to" zoom).
     if ges.direction == "vertical" then
-        inc = ges.distance / math.min(Screen:getHeight(), self._image_wg:getCurrentHeight())
+        local img_h = self._image_wg:getCurrentHeight()
+        local screen_h = Screen:getHeight()
+        if (0.9 * ges.span) > img_h then
```

Comment:
We don't need these parens - it makes it harder to read as I focus on why  it is used (there must be a `-` or a `+` that I don't see :)

Reply:
It's irrelevant here, but in some contexts, parens ensure literals are computed at compile-time instead of runtime, so it's good practice to keep 'em around.

Reply:
I distinctly remember my linter complaining about that one: https://github.com/koreader/koreader/blob/45e4f38c41bdbc476c1a3db38c4b0b7a202c902b/frontend/ui/uimanager.lua#L1302

But I can't find a definitive answer as to how Lua/LuaJIT does constant folding, actually, so let's not worry overmuch about it ;).